juniper_rocket/src/lib.rs

```diff
@@ -107,6 +107,19 @@ where
             ),
         }
     }
+
+    pub fn operation_name(&self) -> Option<&str> {
+        match self {
+            GraphQLBatchRequest::Single(req) => req.operation_name(),
+            GraphQLBatchRequest::Batch(reqs) => {
+                if reqs.len() == 1 {
+                    reqs.get(0).and_then(|req| req.operation_name())
+                } else {
+                    None
```

Comment:
Wouldn't we want a list of operations for batch requests for the same usecase?

Reply:
Yeah that is true. Fixed in https://github.com/graphql-rust/juniper/pull/353/commits/168a763253476d4cc72ac20a8d2b03dab89dfd94